### PR TITLE
refactor: remove axios

### PR DESCRIPTION
**Pull request by benjlevesque**
## Motivation
- reduce size (minor impact)
- avoid versions conflicts
- reduce maintenance

Changing to MSW for mocks also has the benefit to be compatible with any client.

## Choices
- Dropping `gas-price-definer`. This is never used, since #1117 
- dropping `IpfsStore.read`. This is never used. 
- remove `maxSize` from `IpfsManager.read`. This method is kept as a convenience, as it's useful to test the `read` after upload, but is no longer used (since #1117). 
- `axios` remains a `devDependency` of `smart-contracts` and `toolbox` for now

### CODE REVIEW

File: packages/request-node/src/request/getChannelsByTopic.ts [hunk 1]
```diff
@@ -28,6 +28,7 @@ export default class GetChannelHandler {
     clientRequest.setTimeout(GET_CHANNELS_TIMEOUT);
 
     const { updatedBetween, topic } = clientRequest.query;
+    console.log(clientRequest.query);
```
Comment (benjlevesque) [inline packages/request-node/src/request/getChannelsByTopic.ts:31]:
```suggestion
```

File: packages/request-node/src/request/persistTransaction.ts [hunk 2]
```diff
@@ -48,6 +48,7 @@ export default class PersistTransactionHandler {
       !clientRequest.body.transactionData ||
       !clientRequest.body.channelId
     ) {
+      console.log('persist', clientRequest.body);
```
Comment (benjlevesque) [inline packages/request-node/src/request/persistTransaction.ts:51]:
```suggestion
```

File: packages/request-node/test/persistTransaction.test.ts [hunk 3]
```diff
@@ -76,21 +73,4 @@ describe('persistTransaction', () => {
       .set('Accept', 'application/json')
       .expect(StatusCodes.INTERNAL_SERVER_ERROR);
   });
-
-  it('should catch IPFS timeout error', async () => {
```
Comment (benjlevesque) [inline packages/request-node/test/persistTransaction.test.ts:80]: I decided to remove this test. it's a bit hard to replicate and I feel it has no real value

File: packages/ethereum-storage/test/ipfs-manager.test.ts [hunk 4]
```diff
@@ -128,23 +119,27 @@ describe('Ipfs manager', () => {
...
+    );
+    mockServer.listen();
+
+    await expect(ipfsManager.read(hash)).rejects.toThrowError('Failed to fetch');
+    expect(mock).toHaveBeenCalledTimes(retryTestErrorHandling.maxRetries + 1);
+    mockServer.close();
   });
 
-  it('timeout errors should generate retry', async () => {
+  fit('timeout errors should generate retry', async () => {
```
Comment (benjlevesque) [inline packages/ethereum-storage/test/ipfs-manager.test.ts:136]:
```suggestion
  it('timeout errors should generate retry', async () => {
```

File: packages/request-client.js/test/index.test.ts [hunk 5]
```diff
@@ -181,20 +179,18 @@ describe('request-client.js', () => {
       transactions: [TestData.timestampedTransactionWithoutPaymentInfo],
     };
     it('specify the Request Client version in the header', async () => {
-      const mock = new AxiosMockAdapter(axios);
-
-      const callback = (config: any): any => {
-        expect(config.headers[httpConfigDefaults.requestClientVersionHeader]).toBe(
-          packageJson.version,
-        );
```
Comment (benjlevesque) [inline packages/request-client.js/test/index.test.ts:189]: I ditched that test, I don't really see the intention

**Comment by yomarion** [comment]
FYI @benjlevesque, the `yarn.lock` file still references `axios@0.27.2` that looks unused.

File: packages/request-client.js/test/index.test.ts [hunk 6]
```diff
@@ -369,15 +339,34 @@ describe('request-client.js', () => {
...
+      mockServer.events.removeAllListeners();
+      mockServer.resetHandlers();
+      mockServer.close();
+    });
+    beforeEach(() => {
+      hits = { get: 0, post: 0 };
+    });
+    afterEach(() => {
+      // mock.resetHandlers();
+    });
```
Comment (alexandre-abrioux) [inline packages/request-client.js/test/index.test.ts:368]:
```suggestion
```

**Review by alexandre-abrioux — APPROVED** [review]
nice! ✨

File: packages/request-node/test/persistTransaction.test.ts (hunk 3 above)
Comment (MantisClone) [reply packages/request-node/test/persistTransaction.test.ts:80]: I disagree with this. The Request Node still writes to IPFS but it should fail gracefully if for some reason that write fails.

File: packages/request-client.js/test/index.test.ts [hunk 7]
```diff
@@ -181,22 +183,27 @@ describe('request-client.js', () => {
...
+    beforeEach(() => {
+      spyPersistTransaction.mockReturnValue({});
+      spyGetTransactionsByChannelId.mockReturnValue({ result: mockedTransactions });
+    });
+    afterAll(() => {
+      mockServer.close();
+    });
 
+    it('specify the Request Client version in the header', async () => {
```
[the remark this answers is not in the document]
Comment (alexandre-abrioux) [reply packages/request-client.js/test/index.test.ts]: I think it was a typo and it was renamed back to the original name. cc @benjlevesque can you confirm?

[the remark this answers is not in the document]
Comment (benjlevesque) [reply packages/request-client.js/test/index.test.ts]: yes, sorry my comment remained "pending":

good catch. Probably a bad merge or copy paste, I [removed](https://github.com/RequestNetwork/requestNetwork/pull/1237/files#r1395759578) that test

but in the meantim I also addded back a check on the version header

**Pull request merged**